File: internal/services/resend/store.go
package resend

import corestore "github.com/vercel-labs/emulate/internal/core/store"

type Store struct {
	Emails    *corestore.Collection
	Domains   *corestore.Collection
	APIKeys   *corestore.Collection
	Audiences *corestore.Collection
	Contacts  *corestore.Collection
}

func NewStore(runtimeStore *corestore.Store) Store {
	return Store{
		Emails:    runtimeStore.MustCollection("resend.emails", "uuid"),
		Domains:   runtimeStore.MustCollection("resend.domains", "uuid", "name"),
		APIKeys:   runtimeStore.MustCollection("resend.api_keys", "uuid"),
		Audiences: runtimeStore.MustCollection("resend.audiences", "uuid", "name"),
		Contacts:  runtimeStore.MustCollection("resend.contacts", "uuid", "audience_id"),
	}
}

File: packages/@emulators/resend/src/store.ts
import { Store, type Collection } from "@emulators/core";
import type {
  ResendEmail,
  ResendDomain,
  ResendApiKey,
  ResendAudience,
  ResendContact,
  ResendIdempotencyRecord,
  ResendSendAttempt,
} from "./entities.js";

export const ALLOWLIST_DATA_KEY = "allowlist";

export interface ResendStore {
  emails: Collection<ResendEmail>;
  idempotencyKeys: Collection<ResendIdempotencyRecord>;
  sendAttempts: Collection<ResendSendAttempt>;
  domains: Collection<ResendDomain>;
  apiKeys: Collection<ResendApiKey>;
  audiences: Collection<ResendAudience>;
  contacts: Collection<ResendContact>;
}

export function getResendStore(store: Store): ResendStore {
  return {
    emails: store.collection<ResendEmail>("resend.emails", ["uuid"]),
    idempotencyKeys: store.collection<ResendIdempotencyRecord>("resend.idempotency_keys", ["idempotency_key"]),
    sendAttempts: store.collection<ResendSendAttempt>("resend.send_attempts"),
    domains: store.collection<ResendDomain>("resend.domains", ["uuid", "name"]),
    apiKeys: store.collection<ResendApiKey>("resend.api_keys", ["uuid"]),
    audiences: store.collection<ResendAudience>("resend.audiences", ["uuid"]),
    contacts: store.collection<ResendContact>("resend.contacts", ["uuid", "audience_id"]),
  };
}
